src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAICitizenChild.java

```diff
@@ -0,0 +1,311 @@
+    private BlockPos followStart;
+
+    /**
+     * The blockpos the child is visiting
+     */
+    private BlockPos visitHutPos;
+
+    /**
+     * The path for visiting the hut
+     */
+    private PathResult visitingPath;
+
+    /**
+     * Instantiates this task.
+     *
+     * @param citizen the citizen.
+     */
+    public EntityAICitizenChild(@NotNull final EntityCitizen citizen)
+    {
+        super();
+        this.child = citizen;
+        this.setMutexBits(1);
+        stateMachine = new TickRateStateMachine(State.IDLE, this::handleAIException);
+        stateMachine.addTransition(new AIEventTarget(AIBlockingEventType.STATE_BLOCKING, this::updateTimers, stateMachine::getState, 1));
+        stateMachine.addTransition(new AIEventTarget(AIBlockingEventType.EVENT, this::tryGrowUp, () -> State.IDLE, 500));
+
+        stateMachine.addTransition(new AITarget(State.IDLE, this::searchEntityToFollow, () -> State.FOLLOWING, 150));
+        stateMachine.addTransition(new AITarget(State.IDLE, this::isReadyForActivity, () -> State.VISITING, 300));
+
+        stateMachine.addTransition(new AITarget(State.FOLLOWING, this::followingEntity, 20));
+        stateMachine.addTransition(new AITarget(State.VISITING, this::visitHuts, 120));
+    }
+
+    /**
+     * Exception handler for the statemachine
+     */
+    private void handleAIException(final RuntimeException ex)
+    {
+        stateMachine.reset();
+        Log.getLogger().warn("EntityAICitizenChild Child: " + child.getName() + " threw an exception:", ex);
+    }
+
+    /**
+     * Updates all timers
+     */
+    private boolean updateTimers()
+    {
+        // Timer used for delays on actions
+        if (actionTimer > 0)
+        {
+            actionTimer -= Configurations.gameplay.updateRate;
+        }
+
+        AIActiveTime += Configurations.gameplay.updateRate;
+
+        return false;
+    }
+
+    /**
+     * Whether the child moves to a new activity
+     */
+    private boolean isReadyForActivity()
+    {
+        return actionTimer <= 0;
+    }
+
+    /**
+     * Sets the delay till the next activity can start
+     */
+    private void setDelayForNextAction()
+    {
+        // Delay next activity by 3-5min
+        actionTimer = rand.nextInt(2 * 60 * 20) + 3 * 60 * 20;
+    }
+
+    /**
+     * Follow activity preconditions
+     * Search for someone to follow around
+     */
+    private boolean searchEntityToFollow()
+    {
+        if (!isReadyForActivity())
+        {
+            return false;
+        }
+
+        CompatibilityUtils.getWorld(child)
+          // Search entities in radius
+          .getEntitiesInAABBexcluding(
+            child,
+            child.getEntityBoundingBox().expand(
+              (double) START_FOLLOW_DISTANCE,
+              1.0D,
+              (double) START_FOLLOW_DISTANCE),
+            // Limit entity classes
+            target -> target.isEntityAlive() && (target instanceof EntityAgeable || target instanceof EntityPlayer))
```

Comment:
Why EntityAgeable?

That also includes, zombies, villagers and other mods which grow up. I do not think that that is the intended behaviour.

Reply:
Are zombies ageable, I only thought villagers and cows, pigs etc.
I jumped over this in my review. Maybe restrict it to player, villagers and other citizens for now, could add a config for additional things later.

Reply:
I thought ageable would be nice, so childs follow around stuff like sheeps/pigs/other mod critters etc out of curiosity. As you know childs are very curious and interested in everything that moves around them! Was intentional to make it also look at other mod stuff to give it some neat little interaction. I've checked the zombies and other mobs and they dont inherit Ageable.

Reply:
Okay sounds good